### DISCUSSION TITLE
Q: Migration from Disqus Not Working.

### DISCUSSION THREAD

**Discussion by leimao** [walinejs/waline#2267] [Q&A] [answered]
### Describe the bug | 问题描述

Hello,

I was testing migrating the comments from Disqus to Waline using the [Waline Migration Helper](https://waline.js.org/en/migration/tool.html). 

The test data I used was from the [Disqus official website](https://help.disqus.com/en/articles/1717164-comments-export), specifically,

```
<?xml version="1.0"?>
<disqus xmlns="http://disqus.com"
        xmlns:dsq="http://disqus.com/disqus-internals"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://disqus.com/api/schemas/1.0/disqus.xsd
                            http://disqus.com/api/schemas/1.0/internals.xsd"
>

<!--
  Categories must be declared at the top of a file. They only need
  declared once per site, and DISQUS will attempt to validate them       against existing data if they are not present in the XML.

  The `dsq` namespace is for internal usage by DISQUS and tends to
  hold things such as internal identifiers for objects.

- `title` must be unqiue per `forum`
  -->
  <category dsq:id="1">
    <forum>disqusdev</forum>
    <title>Technology</title>
  </category>

<!--
  Threads must be declared after categories, and before posts. They 
  only need declared once per site, and DISQUS will attempt to 
  validate them against existing data if they are not present in the 
  XML.

  The `dsq` namespace is for internal usage by DISQUS and tends to
  hold things such as internal identifiers for objects.

- `id` must be unqiue per `forum`
  -->

  <thread dsq:id="2">
    <id>1</id>
    <forum>disqusdev</forum>
    <category dsq:id="1"/>
    <link/>
    <title/>
    <message/>
    <createdAt>2012-12-12T12:12:12</createdAt>
    <author>
      <name>Baz</name>
    </author>
  </thread>

<!--
  Posts must be declared in a standard tree order. Parents should
  always exist before they are referenced. DISQUS will attempt to 
  validate them against existing data if they are not present in the 
  XML.

  The `dsq` namespace is for internal usage by DISQUS and tends to 
  hold things such as internal identifiers for objects.

- `id` must be unqiue per `forum`
  -->

  <post dsq:id="1">
    <id>2</id>
    <message>Mother Russia</message>
    <thread>1</thread>
    <isSpam>true</isSpam>
    <createdAt>2012-12-12T12:12:12</createdAt>
    <author>
      <name>Baz</name>
    </author>
  </post>
  <post dsq:id="2">
    <id>1</id>
    <message>Yo dude</message>
    <parent dsq:id="1">2</parent>
    <thread>1</thread>
    <createdAt>2012-12-12T12:12:12</createdAt>
    <author>
      <name>Baz</name>
    </author>
  </post>
  <post>
    <id>3</id>
    <message/>
    <thread dsq:id="2"/>
    <createdAt>2012-12-12T12:12:12</createdAt>
    <author>
      <name>Baz</name>
    </author>
  </post>

</disqus>
```

But there is nothing happened after I clicked "Convert". Do you know what's happening? Thanks.

### Website URL | 问题网站

https://waline.js.org/en/migration/tool.html

### Where your waline deploy? | 服务部署在哪里？

Vercel (Default)

### Where your comment data store? | 数据存储在哪里？

LeanCloud(https://leancloud.app)

**Comment by lizheming** [comment, marked as answer]
感谢你的反馈，目前已按照你给的示例数据修复了报错的问题，可以正常导出数据了，你可以再试一下。但是需要注意的是，你提供的示例数据中 `<thread>` 中的 `<link />` 由于没有包含真是的文章地址，所以导出的数据再和页面进行关联的时候，会出现异常。如果真实数据也是类似的情况的话，建议你导出 JSON 文件后，人工不全 `path` 字段后再导入比较好。

---
Thank you for your feedback. The error reporting problem has been fixed according to the sample data you gave, and the data can be exported normally. You can try again. However, it should be noted that the `<link />` in `<thread>` in the sample data you provided does not contain the real article address, so an exception will occur when the exported data is associated with the page. If the real data is in a similar situation, it is recommended that you export the JSON file and manually complete the `path` field before importing it.

**Comment by leimao** [comment]
@lizheming 谢谢你的修复。当我尝试使用自己的 Disqus 数据的时候，却碰到了因为自己的数据过多而数据被截断的问题，请问能否对此进行 relax 或者修复，谢谢。理想情况下，我们是希望能够直接加载一个文件，然后前端  JavaScript 直接进行处理。

---
@lizheming Thanks for the fix. When I tried to use my own Disqus data, I encountered the problem that the data was truncated because I had too much data. Could you please relax or fix this? Thank you. Ideally, we would like to be able to load a file directly and then have the front-end JavaScript process it directly.

**Reply by lizheming** [reply]
可以提供下你的数据我测试下，没有特别明白你说的问题。

---
Can you provide your data and I will test it? I don't understand the problem you are talking about.

**Reply by leimao** [reply]
@lizheming 您好，能够尝试一下下面这个文件吗？谢谢
[tmpSBYnd6.txt](https://github.com/walinejs/waline/files/14004386/tmpSBYnd6.txt)

**Reply by lizheming** [reply]
我测试了，没有问题，这是导出的文件
[output.json](https://github.com/walinejs/waline/files/14004484/output.json)


---
I tested it, there is no problem, this is the exported file
[output.json](https://github.com/walinejs/waline/files/14004484/output.json)

**Reply by leimao** [reply]
Hmm，我刚测了一下也没问题了。不知道是怎么回事，昨天看上去像是有字数限制，无法复制粘贴全部内容。

---
Hmm, I just tested it and it’s fine. I don’t know what happened, but yesterday it looked like there was a character limit and I couldn’t copy and paste the entire content.

**Reply by leimao** [reply]
我建议可以增加一个选择文件的接口。不然如果文件比较大，复制粘贴内容的话有些电脑会卡顿。
但无论怎么说也是非常感谢。

---
I suggest adding an interface for selecting files. Otherwise, if the file is relatively large, some computers will freeze when copying and pasting the content.
But thank you very much anyway.
